src/angular-selector.js

```diff
@@ -6,8 +6,23 @@ export default Selector(complexSelector => {
             throw new Error(`If the selector parameter is passed it should be a string, but it was ${typeof selector}`);
     }
 
-    function getComponentTag (debugElement) {
-        return debugElement.nativeElement.tagName.toLowerCase();
+    validateSelector(complexSelector);
+
+    // NOTE: Angular version 9 or higher
+    const walkingNativeElementsMode = window.ng && typeof window.ng.getComponent === 'function';
+    // NOTE: Angular version 8 or lower
+    const walkingDebugElementsMode = window.ng && typeof window.ng.probe === 'function';
+
+    const isPageReadyForTesting = (walkingNativeElementsMode || walkingDebugElementsMode) &&
+        typeof window.getAllAngularRootElements === 'function';
+
+    if (!isPageReadyForTesting) {
+        throw new Error(`The tested page was built without Angular framework or it didn't load completely.
```

Comment:
@VasilyStrelyaev

Reply:
Checked by Dirk:

The tested page does not use Angular or did not load correctly. Use the 'waitForAngular' function to ensure the page is ready for testing.